linode/instance/helpers.go

```diff
@@ -1368,3 +1369,137 @@ func instanceIPSliceToString(ips []*linodego.InstanceIP) []string {
+) (included bool) {
+	for _, ni := range interfaces1 {
+		included = included || ni.Purpose == linodego.InterfacePurposeVPC
+	}
+
+	for _, ni := range interfaces2 {
+		included = included || ni.Purpose == linodego.InterfacePurposeVPC
+	}
+
+	return included
+}
+
+func BootInstanceAfterVPCInterfaceUpdate(ctx context.Context, meta *helper.ProviderMeta, instanceID, targetConfigID, deadlineSeconds int) diag.Diagnostics {
+	tflog.Debug(ctx, "booting instance after VPC interface change applied")
+	if err := bootInstanceSync(
+		ctx, meta.Client, instanceID, targetConfigID, deadlineSeconds,
+	); err != nil {
+		return diag.Errorf("failed to boot instance after VPC interface change applied: %s", err)
+	}
+	return nil
+}
+
+func ShutdownInstanceForVPCInterfaceUpdate(ctx context.Context, meta *helper.ProviderMeta, instanceID, deadlineSeconds int) diag.Diagnostics {
+	client := &meta.Client
+	if meta.Config.SkipImplicitReboots {
+		return diag.Errorf(
+			"add, remove, and reorder a Linode VPC interface requires implicit " +
+				"reboot of the Linode, please consider setting 'skip_implicit_reboots' " +
+				"to true in the Linode provider config.",
+		)
+	}
+
+	instance, err := client.GetInstance(ctx, instanceID)
+	if err != nil {
+		return diag.Errorf("failed to get instance %d: %s", instanceID, err)
+	}
+
+	tflog.Debug(ctx, "shutting down instance for applying VPC interface change")
+
+	if _, err := waitForRunningOrOfflineState(
+		ctx, instance.Status, client, instance.ID,
+	); err != nil {
+		return diag.Errorf(
+			"failed waiting for instance %d to be in running or offline state: %s", instance.ID, err,
+		)
+	}
+	if instance.Status != linodego.InstanceOffline {
+		if err := shutDownInstanceSync(
+			ctx, *client, instance.ID, deadlineSeconds,
+		); err != nil {
+			return diag.Errorf("failed to shutdown instance: %s", err)
+		}
+	}
+	return nil
+}
+
+func UpdateInterfaces(ctx context.Context, d *schema.ResourceData, meta *helper.ProviderMeta, instanceID, targetConfigID int, managedBoot bool, defaultOpts *linodego.InstanceConfigUpdateOptions) (bool, diag.Diagnostics) {
+	client := meta.Client
+	interfaces := d.Get("interface").([]any)
+
+	expandedInterfaces := helper.ExpandConfigInterfaces(ctx, interfaces)
+	config, err := client.GetInstanceConfig(ctx, instanceID, targetConfigID)
+	if err != nil {
+		return false, diag.Errorf("failed to get config %d: %s", targetConfigID, err)
+	}
+
+	powerOffRequired := VPCInterfaceIncluded(config.Interfaces, expandedInterfaces)
+
+	bootedConfigID, err := helper.GetCurrentBootedConfig(ctx, &client, instanceID)
+	if err != nil {
+		return false, diag.Errorf("failed to get the boot config: %s", err)
+	}
+
+	powerOffRequired = powerOffRequired && bootedConfigID == targetConfigID
+
+	tflog.Debug(ctx, "Updating instance config for interface changes", map[string]any{
+		"config_id": targetConfigID,
+	})
+
+	if err != nil {
+		return false, diag.Errorf("Error fetching data about the current linode: %s", err)
+	}
```

Comment:
I think this error check might be redundant 👍

Reply:
Oops, forgot to clean it up